core/src/main/java/org/jdbi/v3/core/statement/PreparedBatch.java

```diff
@@ -126,8 +108,8 @@ public ResultSetIterable executeAndReturnGeneratedKeys(String... columnNames) {
 
     @Override
     public <R> R execute(ResultProducer<R> producer) {
-        if (parts.isEmpty()) {
-            throw new IllegalStateException("No PreparedBatchParts to execute");
+        if (bindings.isEmpty()) {
```

Comment:
I had the same thought in the shower this morning!  Although after considering it more, I have decided to just `add()` implicitly the final batch if any bindings are present.  If there are any bindings actually missing, the statement will fail to prepare anyway.  If the binding is actually valid, it's unambiguous to just `add()` it, and it feels like we shouldn't `throw` in a situation where we can unambiguously fix the situation...

If you strongly disagree with this decision I'm happy to revisit it.